packages/test/test-end-to-end-tests/src/test/summarizeRefreshLatestSummary.spec.ts

```diff
@@ -53,5 +49,88 @@ describeCompat(
 			await createSummarizer(provider, container, undefined, summaryVersion);
 			await provider.ensureSynchronized();
 		});
+
+		it("The summarizing client will immediately refresh its own summaries", async () => {
+			if (provider.driver.type !== "local") {
+				// skip other drivers as the test would look quite different just to replicate the same behavior
+				return;
+			}
+			const summaryConfig: ISummaryConfiguration = {
+				...DefaultSummaryConfiguration,
+				...{
+					maxTime: 5000 * 12,
+					maxAckWaitTime: 120000,
+					maxOps: 1,
+					initialSummarizerDelayMs: 0,
+				},
+			};
+			const mockLogger = new MockLogger();
+			const testContainerConfig: ITestContainerConfig = {
+				runtimeOptions: {
+					summaryOptions: {
+						summaryConfigOverrides: summaryConfig,
+					},
+				},
+				loaderProps: {
+					logger: mockLogger,
+					configProvider: mockConfigProvider({
+						"Fluid.Summarizer.immediatelyRefreshLatestSummaryAck": true,
+					}),
+				},
+			};
+			const container = await provider.makeTestContainer(testContainerConfig);
+			const container2 = await provider.loadTestContainer(testContainerConfig);
+			const dataObject = (await container.getEntryPoint()) as ITestDataObject;
+			const dataObject2 = (await container2.getEntryPoint()) as ITestDataObject;
+			await waitForContainerConnection(container);
+			await waitForContainerConnection(container2);
+
+			// The first summary needs to be generated otherwise the summarizer will only generate one summary.
+			dataObject._root.set(`first`, "op");
+			await provider.ensureSynchronized();
+			mockLogger.assertMatch([
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+			]);
+			mockLogger.clear();
+
+			const twoSummariesDeferred = new Deferred<void>();
+			const summaryVersions: string[] = [];
+			let summariesSeen = 0;
+			container.on("op", (op) => {
```

Comment:
There is a lot going on here. Can you add a comment explaining what's going on.

Reply:
In order to locally cause two summaries to be generated in a row, I have to do this weird dance, I'm wondering if this is such a freak scenario that it's not happening in production.